### PR TITLE
Separated handling of http and websocket connections.

### PR DESCRIPTION
**Pull request by hauke76**

### CODE REVIEW

File: src/go/cmd/http-relay-server/main_test.go [hunk 1]
```diff
@@ -92,6 +93,7 @@ func TestClientHandler(t *testing.T) {
 
 	wg.Wait()
 	resp := respRecorder.Result()
+	log.Printf(">><<>><< response: %+v", resp)
```
Comment (ensonic) [inline src/go/cmd/http-relay-server/main_test.go:96]: revert

File: src/go/cmd/http-relay-server/utils.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+const (
+	clientPrefix = "/client/"
+)
+
+func addServiceName(span *trace.Span) {
+	relayServerAttr := trace.StringAttribute("service.name", "http-relay-server")
+	span.AddAttributes(relayServerAttr)
+}
+
+func createId() string {
```
Comment (ensonic) [inline src/go/cmd/http-relay-server/utils.go:27]: if this is only used in backend_context.go move it there?

File: src/go/cmd/http-relay-server/utils.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	u := make([]byte, 16)
+	// err is documented as always nil
+	rand.Read(u)
+	return hex.EncodeToString(u)
+}
+
+func extractBackendNameAndPath(r *http.Request) (backendName string, path string, err error) {
```
Comment (ensonic) [inline src/go/cmd/http-relay-server/utils.go:34]: Same here? Move to backend_context.go.

File: src/go/cmd/http-relay-server/utils.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+		Host:   proto.String(r.Host),
+		Url:    proto.String(backendUrl.String()),
+		Header: marshalHeader(&r.Header),
+		Body:   body,
+	}
+
+	return backendReq
+}
+
+func marshalHeader(h *http.Header) []*pb.HttpHeader {
```
Comment (ensonic) [inline src/go/cmd/http-relay-server/utils.go:82]: these 2 could probably even go to a httpproto.go file.

File: src/go/cmd/http-relay-server/handler_http.go
Comment (ensonic) [inline src/go/cmd/http-relay-server/handler_http.go:1]: should this then be handler_single.go ?

**Review by ensonic — APPROVED** [review]

File: src/go/cmd/http-relay-server/handler_http.go
Comment (hauke76) [reply src/go/cmd/http-relay-server/handler_http.go:1]: That's a nice suggestion. I chose _http since I was simply too uncreative.

**Pull request merged**